Compile errors when USE_UNIXSOCK is off

```
$ cmake -DUSE_UNIXSOCK=OFF .
```

```
[ 75%] Linking C executable retest
Undefined symbols for architecture arm64:
  "_unixsock_listen_fd", referenced from:
      _test_unixsock in unixsock.c.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [test/retest] Error 1
make[2]: *** [test/CMakeFiles/retest.dir/all] Error 2
make[1]: *** [test/CMakeFiles/retest.dir/rule] Error 2
make: *** [retest] Error 2
```

--- a/test/sa.c
+++ b/test/sa.c
@@ -22,11 +22,13 @@ int test_sa_cmp(void)
 		uint16_t port2;
 		bool eq;
 	} testv[] = {
+#if HAVE_UNIXSOCK
 		{
 			"unix:/test.sock", 0,
 			"unix:/test.sock", 0,
 			true
 		},
+#endif
 		{
 			"1.2.3.4", 12345,
 			"1.2.3.4", 12345,
@@ -347,7 +349,9 @@ int test_sa_pton(void)
 		{"fa01::2a29",                      0            },
 		{"127.0.0.1",                       0            },
 		{"192.168.110.2",                   0            },
+#if HAVE_UNIXSOCK
 		{"unix:/test.sock",                 0            },
+#endif
 		{"fe80::xxxx:d8d9:ddc3:25dd:%eth0", EADDRNOTAVAIL},
 	};
 

--- a/test/unixsock.c
+++ b/test/unixsock.c
@@ -18,6 +18,7 @@
 #include <re_dbg.h>
 
 
+#if HAVE_UNIXSOCK
 static void http_req_handler(struct http_conn *conn,
 			     const struct http_msg *msg, void *arg)
 {
@@ -25,10 +26,12 @@ static void http_req_handler(struct http_conn *conn,
 	(void)msg;
 	(void)arg;
 }
+#endif
 
 
 int test_unixsock(void)
 {
+#if HAVE_UNIXSOCK
 	struct sa srv;
 	re_sock_t fd = RE_BAD_SOCK;
 	struct http_sock *sock;
@@ -57,4 +60,7 @@ out:
 	if (err)
 		(void)unlink(&socket[5]);
 	return err;
+#else
+	return ESKIPPED;
+#endif
 }

